cmd/kustomize/docs/kustomize.yaml

```diff
@@ -1,15 +1,105 @@
+# the kustomize program.
+#
+# Since customization is, by definition, _custom_,
+# there are no sensible default values for the fields
+# in this file.
+#
+# The field values used below are merely examples, not
+# to be copied literally.  The values won't work if
+# they happen to be references to external files that
+# don't exist.
+#
+# In practice, fields with no value should simply be
+# omitted from kustomize.yaml to reduce the content
+# visible in configuration reviews.
+# ----------------------------------------------------
+
+
+# Value of this field is prepended to the
+# names of all resources, e.g. a deployment named
+# "wordpress" becomes "alices-wordpress".
+namePrefix: alices-
+
+# Labels to add to all resources and selectors.
 labelsToAdd:
-  app: helloworld
+  someName: someValue
+  owner: alice
+  app: bingo
+
+# Annotations (non-identifying metadata)
+# to add to all resources.  Like labels,
+# these are key value pairs.
 annotationsToAdd:
-  note: This is an example annotation
-resources: []
-#- service.yaml
-#- ../some-dir/
-# There could also be configmaps in Base, which would make these overlays
-configMapGenerator: []
-# There could be secrets in Base, if just using a fork/rebase workflow
-secretGenerator: []
+  oncallPager: 800-555-1212
+
+# Each entry in this list must resolve to an existing
+# resource definition in YAML.  These are the resource
+# files that kustomize reads, modifies and emits as a
+# YAML string, with resources separated by document
+# markers ("---").
+resources:
+ - some-service.yaml
+ - ../some-dir/some-deployment.yaml
+
+# Each entry in this list results in the creation of
+# one ConfigMap resource (it's a generator of n maps).
+# The example below creates a ConfigMap with the
+# names and contents of the given files.
+configMapGenerator:
+- name: myJavaServerProps
+  files:
+    - application.properties
+    - more.properties
+
+# Each entry in this list results in the creation of
+# one Secret resource (it's a generator of n secrets).
+# A command can do anything to get a secret,
+# e.g. prompt the user directly, start a webserver to
+# initate an oauth dance, etc.
+secretGenerator:
+- name: app-tls
+  commands:
+    tls.crt: "cat secret/tls.cert"
+    tls.key: "cat secret/tls.key"
+  type: "kubernetes.io/tls"
+
+# Each entry in this list should resolve to a directory
+# containing a kustomization file, else the
+# customization fails.
+#
+# The presence of this field means this file (the file
+# you a reading) is an _overlay_ that further
+# customizes information coming from these _bases_.
+#
+# The customization declarations in these _base_ files
+# are applied first, in order, to be followed by the
+# customizations declared in this file (the _overlay_).
+#
+# Typical use case: a dev, staging and production
+# environment that are mostly identical but differing
+# crucial ways (image tags, a few server arguments,
+# etc. that differ from the common base).
+bases:
+- ../../base
```

Comment:
So obviously nerd-me likes the idea of a list, but UX-me wonders if it's overkill.

As an example:

```
bases:
- ../1
- ../2
```

Does everything in `2` need to reference things in `1`? (if so, I'd think we'd simply have `1` as the base for `2`)

Or are we combining two things that are effectively two packages - e.g. 1 is `apache-wordpress` and 2 is `mysql` ?

Reply:
Brian wanted a list :)
It does seem a little too powerful and confusing.

Reply:
BTW, it's the latter.  Everything from these various bases, regardless of file system hierarchy, gets sucked into the non-hierarchical sea of resources (module name collisions, which is a detected error).